api/src/services/files.ts

```diff
@@ -47,19 +48,27 @@ export class FilesService extends ItemsService {
 	): Promise<PrimaryKey> {
 		const storage = await getStorage();
 
-		let existingFile: Record<string, any> | null = null;
+		let existingFile: Pick<
+			File,
+			'folder' | 'filename_download' | 'filename_disk' | 'title' | 'description' | 'metadata' | 'replaced_on'
+		> | null = null;
 
 		// If the payload contains a primary key, we'll check if the file already exists
 		if (primaryKey !== undefined) {
 			// If the file you're uploading already exists, we'll consider this upload a replace so we'll fetch the existing file's folder and filename_download
 			existingFile =
 				(await this.knex
-					.select('folder', 'filename_download', 'filename_disk', 'title', 'description', 'metadata')
+					.select('folder', 'filename_download', 'filename_disk', 'title', 'description', 'metadata', 'replaced_on')
 					.from('directus_files')
 					.where({ id: primaryKey })
 					.first()) ?? null;
 		}
 
+		if (existingFile) {
+			const helpers = getHelpers(this.knex);
+			existingFile.replaced_on = new Date(helpers.date.writeTimestamp(new Date().toISOString()));
```

Comment:
I had to type `replaced_on` to be a `Date` value, as I was having trouble getting the typescript to work without casting.